src/fondant/pipeline/lightweight_component.py

```diff
@@ -65,11 +68,62 @@ class LightweightComponent(BaseComponent):
     def image(cls) -> Image:
         raise NotImplementedError
 
+    @classmethod
+    def consumes(cls) -> t.Optional[t.Dict[str, t.Any]]:
+        pass
+
+    @classmethod
+    def modify_consumes_spec(cls, apply_consumes, consumes_spec):
+        """Modify fields based on the consumes argument in the 'apply' method."""
+        if apply_consumes:
+            for k, v in apply_consumes.items():
+                if isinstance(v, str):
+                    consumes_spec[k] = consumes_spec.pop(v)
+                else:
+                    msg = (
+                        f"Invalid data type for field `{k}` in the `apply_consumes` "
+                        f"argument. Only string and pa.DataType are allowed."
```

Comment:
If I see it correctly, we are not checking if the type is pa.DataType?

Reply:
good catch only strings are allowed in case no `consumes` is passed on the decorator level